docs/AnnotationProcessors.md

```diff
@@ -0,0 +1,182 @@
+# Annotation Processors
+Annotation Processors (APs) are optional configuration tools that allow user space to define dynamic template content.
+
+## How Annotation Processors Work
+When a template includes an appropriate [annotation][Annotations] (a special comment with a particular tag),
+Buphalo will replace that comment with the results of the specified Annotation Processor.
+If no Annotation Processor is specified, Buphalo uses a default AP that replaces the annotation with the contents of
+the comment (sans annotation tag)
+
+### Template Annotations
+The annotations in a template use the following form with two variables:
+```php
+/** @neighborhoods-buphalo:annotation-processor AnnotationProcessorKey
+Default Contents
+*/
+```
+- `AnnotationProcessorKey`: Also used in the fabrication file to tie the annotation to the specific AP.
+- `Default Contents`: If no AnnotationProcessor is specified in the fabrication file, this is used instead
+- There MUST be a newline between the `AnnotationProcessorKey` and `Default Contents`
```

Comment:
I thought that was the case, but I just tested it and got it to work if I didn't include Default Contents. It may have been fixed since then?
```
    public const TEST_STRING = "/** @neighborhoods-buphalo:annotation-processor TestSimpleStringAnnotationProcessor */";
```
plus
```
      TestSimpleStringAnnotationProcessor:
        processor_fqcn: \Neighborhoods\Buphalo\V1\AnnotationProcessors\SimpleString
        static_context_record:
          string: 'this is a string'
```
produced
```
    public const TEST_STRING = "this is a string";
```

Reply:
The problem actually comes up when you have more than 1 annotation processor in a file. Assuming Buphalo uses that same regex matching as Bradfab, I believe the below example will break. The first AP will replace everything from the first  `/**` after `TEST_STRING = "` to the last `*/` after `ThisWillProbablyDisappearToo`.

```
    public const TEST_STRING = "/** @neighborhoods-buphalo:annotation-processor TestSimpleStringAnnotationProcessor */";

// This will probably disappear

/** @neighborhoods-buphalo:annotation-processor ThisWillProbablyDisappearToo */
```

Reply:
> because the incorrect example works with one AP in a template, but once you add a second AP, things break.

Huh, just re-read this, and that's very interesting. I'll see if I can replicate with some tests.

Reply:
Thanks, @jakemalachowski 

I've been able to validate this with tests. Created [BUPH-92 | Single-Line Annotation Tags Don't Work Well With Others](https://55places.atlassian.net/browse/BUPH-92) and #38 for us to address.